test/strings/search.jl

```diff
@@ -379,3 +379,9 @@ end
 @test rsearchindex("\U1f596\U1f596", "\U1f596\U1f596", endof("\U1f596\U1f596\U1f596")) == 1
 
 @test_throws ErrorException "ab" ∈ "abc"
+
+# issue #15723
```

Comment:
Could be worth testing other cases for each function, e.g. with the searched character is in the middle of the string (with non-ASCII chars around it).